src/V3AstNodes.cpp

```diff
@@ -1502,12 +1502,14 @@ void AstClassExtends::dump(std::ostream& str) const {
     if (isImplements()) str << " [IMPLEMENTS]";
 }
 AstClass* AstClassExtends::classp() const {
-    const AstClassRefDType* refp = VN_CAST(dtypep(), ClassRefDType);
-    if (VL_UNLIKELY(!refp)) {  // LinkDot uses this for 'super.'
-        refp = VN_AS(childDTypep(), ClassRefDType);
+    const AstNodeDType* const dtp = dtypep() ? dtypep() : childDTypep();
+    const AstClassRefDType* const refp = VN_CAST(dtp, ClassRefDType);
+    if (refp && !refp->paramsp()) {
+        // class already resolved
+        return refp->classp();
+    } else {
+        return nullptr;
     }
-    UASSERT_OBJ(refp, this, "class extends non-ref");
-    return refp->classp();
```

Comment:
Previously this would only be called at appropriate points.  I worry using this to also check may make it harder to assert it's called the right place. Thoughts?

Reply:
I think that it's better to have a method that returns nullptr if the object is not resolved and then check the result after the call instead of getting the object manually on stages on which the method may throw an error.
But we can also add another method, that would call this one and assert the result.

Reply:
I added a method that returns the extended class, but doesn't throw error.